File: app/backend/src/main/kotlin/pwr/psi/Powierzenia3000/entrustment/model/Entrustment.kt
package pwr.psi.Powierzenia3000.entrustment.model

import pwr.psi.Powierzenia3000.profile.model.Lecturer
import javax.persistence.*

@Entity
@Table(name = "entrustment")
data class Entrustment(
        @Id @GeneratedValue
        val id: Long = 0,
        val answer: String? = "",
        val entrustmentStatus: EntrustmentStatus = EntrustmentStatus.NEW,
        val hours: Double? = 0.0,
        //TODO
        @ManyToOne
        @JoinColumn(name = "lecturer_id")
        val lecturer: Lecturer? = null,
        @ManyToOne
        @JoinColumn(name = "course_id")
        val courseId: Course? = null
)


enum class EntrustmentStatus {
    NEW, REJECTED, ACCEPTED, PROPOSED
}

File: app/frontend/src/shared/models/Entrustment.ts
import { Lecturer } from "./Lecturer";
import { Course } from "./Course";

export interface Entrustment {
  id: number;
  answer: string;
  entrustmentStatus: string;
  hours: number;
  lecturers: Array<Lecturer>;
  courseId: Course;
}

export enum EntrustmentStatus {
  NEW = "Nowe",
  REJECTED = "Odrzucone",
  ACCEPTED = "Zaakceptowane",
  PROPOSED = "Zaproponowane"
}

export function mapEntrustmentStatus(status: EntrustmentStatus): string {
  switch (status) {
    case EntrustmentStatus.ACCEPTED:
      return "ACCEPTED";
    case EntrustmentStatus.REJECTED:
      return "REJECTED";
    case EntrustmentStatus.NEW:
      return "NEW";
    case EntrustmentStatus.PROPOSED:
      return "PROPOSED";
    default:
      break;
  }
}
